collectors/freeipmi.plugin/freeipmi_plugin.c

```diff
@@ -645,959 +711,694 @@ static int excluded_status_record_ids_check(int record_id) {
+                sensor_name, record_id, sensor_number, sensor_type, sensor_state, sensor_units, sensor_reading_type);
 
-#ifdef IPMI_MONITORING_SENSOR_READING_FLAGS_ENTITY_SENSOR_NAMES
-    if (entity_sensor_names)
-        sensor_reading_flags |= IPMI_MONITORING_SENSOR_READING_FLAGS_ENTITY_SENSOR_NAMES;
-#endif // IPMI_MONITORING_SENSOR_READING_FLAGS_ENTITY_SENSOR_NAMES
+    // check if it is excluded
+    bool excluded_metric = excluded_record_ids_check(record_id);
+    bool excluded_state = excluded_status_record_ids_check(record_id);
 
-    if (!record_ids_length && !sensor_types_length)
-    {
-        if ((sensor_count = ipmi_monitoring_sensor_readings_by_record_id (ctx,
-                hostname,
-                ipmi_config,
-                sensor_reading_flags,
-                NULL,
-                0,
-                NULL,
-                NULL)) < 0)
-        {
-            collector_error( "ipmi_monitoring_sensor_readings_by_record_id(): %s",
-                             ipmi_monitoring_ctx_errormsg (ctx));
-            goto cleanup;
-        }
-    }
-    else if (record_ids_length)
-    {
-        if ((sensor_count = ipmi_monitoring_sensor_readings_by_record_id (ctx,
-                hostname,
-                ipmi_config,
-                sensor_reading_flags,
-                record_ids,
-                record_ids_length,
-                NULL,
-                NULL)) < 0)
-        {
-            collector_error( "ipmi_monitoring_sensor_readings_by_record_id(): %s",
-                             ipmi_monitoring_ctx_errormsg (ctx));
-            goto cleanup;
-        }
+    if(excluded_metric) {
+        if(state->debug)
+            fprintf(stderr, "Sensor '%s' is excluded by excluded_record_ids_check()\n", sensor_name);
     }
-    else
-    {
-        if ((sensor_count = ipmi_monitoring_sensor_readings_by_sensor_type (ctx,
-                hostname,
-                ipmi_config,
-                sensor_reading_flags,
-                sensor_types,
-                sensor_types_length,
-                NULL,
-                NULL)) < 0)
-        {
-            collector_error( "ipmi_monitoring_sensor_readings_by_sensor_type(): %s",
-                             ipmi_monitoring_ctx_errormsg (ctx));
-            goto cleanup;
-        }
+
+    if(excluded_state) {
+        if(state->debug)
+            fprintf(stderr, "Sensor '%s' is excluded for status check, by excluded_status_record_ids_check()\n", sensor_name);
     }
 
-#ifdef NETDATA_COMMENTED
-    printf ("%s, %s, %s, %s, %s, %s, %s, %s, %s, %s\n",
-            "Record ID",
-            "Sensor Name",
-            "Sensor Number",
-            "Sensor Type",
-            "Sensor State",
-            "Sensor Reading",
-            "Sensor Units",
-            "Sensor Event/Reading Type Code",
-            "Sensor Event Bitmask",
-            "Sensor Event String");
-#endif // NETDATA_COMMENTED
```

Comment:
And finally another one here:

```sh
*** CID 462563:  Control flow issues  (DEADCODE)
/collectors/freeipmi.plugin/freeipmi_plugin.c: 906 in netdata_update_ipmi_sensor_reading()
900     
901         if(excluded_state) {
902             if(state->debug)
903                 fprintf(stderr, "Sensor '%s' is excluded for status check, by excluded_status_record_ids_check()\n", sensor_name);
904         }
905     
>>>     CID 462563:  Control flow issues  (DEADCODE)
>>>     Execution cannot reach the expression """" inside this statement: "t.sensor_name = strdupz((se...".
906         struct sensor t = {
907                 .sensor_type = sensor_type,
908                 .sensor_state = sensor_state,
909                 .sensor_units = sensor_units,
910                 .sensor_reading_type = sensor_reading_type,
911                 .sensor_name = strdupz(sensor_name ? sensor_name : "")
```

Reply:
I am going to test compilation on FreeBSD now, but we need to verify whether these reports as false positive or not.

Reply:
Also for compatibility. Ignore.